File: Leetcode_POTD_2025/7_JULY/1957_Delete_Characters_to_Make_Fancy_String.cpp
class Solution {
public:
    std::string makeFancyString(std::string s) {
        std::string astral_result;
        char last_orb = '\0';
        int orb_count = 0;

        for (char current_orb : s) {
            if (current_orb == last_orb) {
                orb_count++;
            } else {
                last_orb = current_orb;
                orb_count = 1;
            }

            if (orb_count <= 2) {
                astral_result.push_back(current_orb);
            }
        }
        return astral_result;
    }
};

File: Leetcode_POTD_2025/7_JULY/1957_Delete_Characters_to_Make_Fancy_String.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder astralResult = new StringBuilder();
        char lastOrb = '\0';
        int orbCount = 0;

        for (char currentOrb : s.toCharArray()) {
            if (currentOrb == lastOrb) {
                orbCount++;
            } else {
                lastOrb = currentOrb;
                orbCount = 1;
            }

            if (orbCount <= 2) {
                astralResult.append(currentOrb);
            }
        }
        return astralResult.toString();
    }
}
